src/supervisor.rs

```diff
@@ -0,0 +1,614 @@
+use std::collections::HashMap;
+use std::sync::Arc;
+
+use crate::context::{self, Context};
+use crate::events::EventNotifier;
+use crate::worker::{self, Worker};
+
+#[derive(Clone, PartialEq, Debug)]
+pub enum Order {
+    LeftToRight,
+    RightToLeft,
+}
+
+#[derive(Clone, PartialEq, Debug)]
+pub enum Strategy {
+    OneForOne,
+}
+
+#[derive(Debug)]
+pub struct StartError {
+    pub failing_child_runtime_name: String,
+    pub failing_child_error: Arc<worker::StartError>,
+    pub failing_sibling_termination_error: Option<TerminationError>,
+}
+
+#[derive(Debug)]
+pub struct TerminationError(HashMap<String, Arc<worker::TerminationError>>);
+
+pub struct Spec {
+    pub name: String,
+    pub ev_notifier: EventNotifier,
+    pub order: Order,
+    pub strategy: Strategy,
+    pub children: Vec<worker::Spec>,
+}
+
+pub struct Supervisor {
+    pub runtime_name: String,
+    pub children: Vec<Worker>,
+
+    pub name: String,
+    pub ev_notifier: EventNotifier,
+    pub order: Order,
+    pub strategy: Strategy,
+}
+
+fn ordered_children_on_start(
+    order: &Order,
```

Comment:
Since `Order` is a two variant `Clone`-able enum, it is actually more efficient to pass it by value than by reference.
```suggestion
    order: Order,
```